src/main/webapp/assets/js/productpage.js

```diff
@@ -59,36 +74,53 @@ function createCategoryElement(categoryName) {
 }
 
 function createThumbnailImageElement(imageUrl) {
-  return `<div class="lightbox-column">
-          <img
-            src=${imageUrl}
-            style="height: 250px; width:auto;"
-            onclick="openModal();currentSlide(1)"
-            class="hover-shadow cursor"
-          />
-          `;
+  var thumbnailImageElement = document.createElement("div");
+  thumbnailImageElement.className = "lightbox-column";
+  thumbnailImageElement.innerHTML = `<img
+                                        src=${imageUrl}
+                                        style="height: 250px; width:auto;"
+                                        onclick="openModal();currentSlide(1)"
+                                        class="hover-shadow cursor"
+                                      />`
+  return thumbnailImageElement;
 }
 
 function createGalleryImageElement(imageUrl, index, length) {
-  return `<div class="mySlides">
-              <div class="numbertext">${index} / ${length}</div>
-              <img
-                src=${imageUrl}
-                style="height: 400px; width:auto;"
-              />
-            </div>
-          `;
+  var slide = document.createElement("div");
+  slide.className = "mySlides";
+  slide.innerHTML = `<div class="numbertext">${index} / ${length}</div>
+                      <img
+                        src=${imageUrl}
+                        style="height: 400px; width:auto;"
+                      />
+                    `;
+  return slide;
 }
 
 function createGalleryImagePreviewElement(imageUrl, index) {
-  return `<div class="lightbox-column">
-              <img
-                class="demo cursor"
-                src=${imageUrl}
-                style="height: 100px; width:auto"
-                onclick="currentSlide(${index})"
-                alt="Sea Salt Brownies"
-              />
-            </div>
-          `;
+  var galleryImagePreviewElement = document.createElement("div");
+  galleryImagePreviewElement.className = "lightbox-column";
+  galleryImagePreviewElement.innerHTML = `<img
+                                            class="demo cursor"
+                                            src=${imageUrl}
+                                            style="height: 100px; width:auto"
+                                            onclick="currentSlide(${index})"
+                                            alt="Sea Salt Brownies"
+                                          />`;
+  return galleryImagePreviewElement;
+}
+
+function createEditBusinessLink(isAdmin) {
+  var editLink = document.getElementById("editBusinessLink");
+  if (isAdmin !== "true") {
+    editLink.parentNode.removeChild(editLink);
+    return;
+  }
+  
+  const queryString = window.location.search;
+  const urlParams = new URLSearchParams(queryString);
+  const businessId = urlParams.get("businessID");
+  editLink.href = `edit.html?businessID=${businessId}`;
```

Comment:
When I clicked on Edit This Business I got this 

/edit.html?businessID=null

as the url

this is because on the product page itself the url had no businessID parameters, only /product.html

Reply:
This is happening only for the featured listing when I click "Click to Find Out More". The others are linking up fine.

Reply:
ah it's because the featured listing is hardcoded, so I couldn't populate the businessID url parameter. might need to talk to @jacoblipech about randomly picking a business to show

Reply:
let me change the logic check so Edit This business doesn't show for pages without a businessID parameter @dalisc